src/lib.rs

```diff
@@ -289,19 +300,29 @@ impl Calc {
                     block,
                     alternative,
                 } => {
-                    if let Ok(Some(StackValue::Boolean(val))) = self.eval(condition, scope) {
+                    if let Ok(Some(EvalResult::Value(StackValue::Boolean(val)))) =
+                        self.eval(condition, scope)
+                    {
+                        let mut block_result = None;
                         if val {
-                            self.eval(block, scope)?;
+                            block_result = self.eval(block, scope)?;
                         } else if let Some(alt) = alternative {
-                            self.eval(alt, scope)?;
+                            block_result = self.eval(alt, scope)?;
+                        }
+                        if let Some(EvalResult::Jump(JumpInstruction::Return)) = block_result {
+                            break;
                         }
                     }
                 }
             }
         }
+        let result;
         if self.stack.is_empty() {
-            return Ok(None);
+            result = Ok(None);
+        } else {
+            result = Ok(Some(EvalResult::Value(self.stack.pop().unwrap())));
```

Comment:
do we know this unwrap can't fail?

Reply:
I'm not sure cause its been there for a while. I'll add a check for it

Reply:
That's actually why I added the `.pop` wrapper function.  Changed :point_right: 85a8e361e54a404ab3c4ccc095538361715eee72